### DISCUSSION TITLE
HSTS (and other security headers) for GitHub Pages with custom domains

### DISCUSSION THREAD

**Discussion by haliphax** [community/community#4444] [Other Feature Feedback, Questions, & Ideas] [answered]
What it says on the tin. Would be nice to be able to send HSTS (and other security headers) for sites on custom domains hosted by GitHub Pages. An immediate failure for the Security category is encountered when using scanners like webpagetest.org, etc.

Seems to have something to do with the age of my account.

There was an issue for this on the old "github" repo, but it doesn't look like it had been revived as a discussion here since.

<s>Edit: So... apparently, there is a more recent request for this, but it's on a separate "community" site? I am confused.

https://github.community/t/feature-request-hsts-for-github-pages/176265</s>

Edit: Looks like the other request has vanished.

**Comment by TPS** [comment]
As a note, a major thread preceding this 1 is https://github.com/isaacs/github/issues/1249.

**Reply by haliphax** [reply]
Yep, started there. :smile:

**Comment by sorat0mo** [comment]
This is the only thing stopping me moving hosting my tiny blog and project sites from self-hosted to Pages.

**Comment by AlFalahTaieb** [comment]
This year is the year.

**Reply by tvdijen** [reply]
For suck's sake Microsoft, we need safe Github pages. Start implementing proper support for .htaccess files

**Reply by tvdijen** [reply]
@Rabadash8820 Care to discuss why not?

**Reply by Rabadash8820** [reply]
@tvdijen I downvoted your tone, not your request

**Reply by tvdijen** [reply]
Get a grip!  The way they're dealing with this is a big joke.. What ever happened to 'security by design'?

**Comment by vedranmiletic** [comment]
There seems to be a development in this direction (not configurable, at least yet):

``` shell
% fetch -o - -vv https://group.miletic.net/ | & grep Security
<<< Strict-Transport-Security: max-age=31556952
```

**Reply by haliphax** [reply]
Not seeing that on my own domain, <s>but this is good news!</s>

**Reply by MatTheCat** [reply]
Same for me, no `Strict-Transport-Security` on my domain. @vedranmiletic maybe your GitHub account was created after June 15, 2016 (cf. https://github.com/isaacs/github/issues/1249#issuecomment-872428119)?

**Reply by vedranmiletic** [reply]
@MatTheCat my account wasn't, but the organization indeed was. That's good to know.

**Comment by meysam81** [comment]
2024, any update here? Simply add the header to the response :shrug: 
Or let us add the header in the settings tab

**Reply by haliphax** [reply]
The issue is nearly 6 years old at this point if you count the original post: https://github.com/isaacs/github/issues/1249

I wouldn't hold my breath. If this feature is ever implemented, it will have nothing to do with users requesting it.

**Reply by tvdijen** [reply]
Ffing ridiculous

**Reply by haliphax** [reply]
The only thing that you can do in the meantime is create a new user/org and migrate your repository, as it seems to be present in the headers for newer accounts.

**Reply by tvdijen** [reply]
😮 You serious?!  I'm not really willing to move 80 repo's within my org though..  They are all composer-repo's, meaning that I would have to re-register them in Packagist (and add an alias to it for BC)... Not really something I'd like to spend my limited time on.

I hope it is an indicator that this functionality is finally becoming generally available on the short term.

**Reply by haliphax** [reply]
It's not. It's been that way since before I posted this discussion. Users/orgs created after some specific date (in 2020 I think? not sure) don't have this issue.

**Reply by tvdijen** [reply]
That's just fucked up

**Reply by meysam81** [reply]
> Users/orgs created after some specific date (in 2020 I think? not sure) don't have this issue.

I have created my GitHub Pages a couple of weeks ago. And I don't have HSTS 🤷

```bash
curl -sSfLo /dev/null https://developer-friendly.blog -D - | grep -i security
# nothing
```

**Reply by haliphax** [reply]
But how old is the account/org the repo is owned by?

**Reply by haliphax** [reply]
https://github.com/isaacs/github/issues/1249#issuecomment-872428119

**Reply by meysam81** [reply]
Yes, the account is also created a few months ago (end of 2023).

**Reply by haliphax** [reply]
Interesting. No idea, then. :shrug:

**Reply by tvdijen** [reply]
Oh well, I have bugs reported to Microsoft 2yrs ago that still haven't been fixed.. It's ridiculous, but we kind of have to deal with them...

**Comment by ElioDiNino** [comment, marked as answer]
As I first saw in https://github.com/orgs/community/discussions/113716#discussioncomment-12344699 and recently confirmed myself, you can contact GitHub Support, and they can manually add the HSTS header to sites of your choosing.
